core/trino-spi/src/main/java/io/trino/spi/classloader/ThreadContextClassLoader.java

```diff
@@ -13,9 +13,24 @@
  */
 package io.trino.spi.classloader;
 
+import java.io.Closeable;
+
 public class ThreadContextClassLoader
+        implements Closeable
 {
-    private ThreadContextClassLoader() {}
+    private final ClassLoader originalThreadContextClassLoader;
+
+    public ThreadContextClassLoader(ClassLoader newThreadContextClassLoader)
```

Comment:
Please mark it as `@Deprecated` and write what should be used instead.

Reply:
Or maybe we could hide constructor here?

Reply:
added `@Deprecated` annotation

Reply:
@wendigo How to hide the constructor? Any change to access modifiers would be breaking change of the SPI.

Reply:
@sergey-melnychuk you are right :)